test/network_generation/test_dataflow_dynamics.rb

```diff
@@ -360,6 +363,151 @@ module NetworkGeneration
                     add_agents(tasks[0, 2])
                     flexmock(@dynamics).should_receive(:propagate).with(tasks[0, 2])
                 end
+
+                it "handles the case where the explicit policy sets the type to :data" do
```

Comment:
And what about the type being `:buffer`. We should make sure it's not overriding the size.